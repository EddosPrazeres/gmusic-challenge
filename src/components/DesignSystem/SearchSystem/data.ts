export interface ISearchSystem {   
  title?: string;
  titleButton: string;  
}

export interface ISearchSystemLayout extends ISearchSystem {  
  value: string | ReadonlyArray<string> | number;
  onChange: (e: any) => void;
  onSubmit: (e: any) => void;
}

type tplotOptions = {
  [key: string]: string
}

export const FontSize: tplotOptions = {
  tiny: '10px',
  small: '13px',  
  medium: '16px',
  large: "18px"
}
