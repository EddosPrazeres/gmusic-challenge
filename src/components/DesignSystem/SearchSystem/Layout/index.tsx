import * as React from "react";
import { Text } from "../../../../components";
import { ISearchSystemLayout } from "../data";
import './styles.scss'

export function SearchSystem({ 
  value,
  title,
  titleButton,
  onSubmit,
  onChange,
  ...props 
}: ISearchSystemLayout) {

  return (
    <form className="SearchSystem" onSubmit={onSubmit}>
      {title && <Text title={title} size="large"/> }
        <div className="SearchSystem-content"> 
          <input type="text" value={value} onChange={onChange} />      
          <input type="submit" value={titleButton} />
        </div>
    </form>
  )
}
