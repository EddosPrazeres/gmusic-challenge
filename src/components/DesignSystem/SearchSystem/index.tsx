import React, { useEffect, useState} from 'react';
import { ISearchSystem } from './data';
import { SearchSystem as Layout } from './Layout';
import { useUser } from '../../../hooks'

export default function SearchSystem({...props}: ISearchSystem) {
  const [value, setValue] = useState("");
  const { getSpecificUser, clearSpecificUser } = useUser()

  useEffect(()=> {
    clearSpecificUser(value)
  }, [value, clearSpecificUser])

  const onChange = (e: any) => {
    setValue(e.target.value);
  }
  const onSubmit = (e: any) => {
    e.preventDefault();
    getSpecificUser(value);
  }

  return (
    <Layout {...props} value={value} onChange={onChange} onSubmit={onSubmit}/>
  )
}

  