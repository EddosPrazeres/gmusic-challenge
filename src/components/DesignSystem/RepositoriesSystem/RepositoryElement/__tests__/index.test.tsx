import React from 'react'
import { mount, shallow } from 'enzyme'
import SearchSystem from '../index'

const createComponent = (props, useDefault = false) => {
  const componentProps = {    
    ...props,
  }

  const wrapper = useDefault ? mount(<SearchSystem  {...componentProps} />) : shallow(<SearchSystem     {...componentProps} />)
  const unmount = () => wrapper.unmount()

  return [wrapper, unmount]
}

describe('SearchSystem', () => {
  it('should match snapshot', () => {
    const [wrapper, unmount] = createComponent({})

    expect(wrapper).toMatchSnapshot()
    unmount()
  })
})
