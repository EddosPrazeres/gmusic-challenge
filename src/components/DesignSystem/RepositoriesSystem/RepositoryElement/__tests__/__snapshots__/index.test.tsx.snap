// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchSystem should match snapshot 1`] = `ShallowWrapper {}`;
